### ISSUE TITLE
Add commands to deal with storage

### ISSUE THREAD

**Issue by luizirber** [dib-lab/sourmash#261]
- `prefetch`: download all nodes beforehand (useful for IPFS)
- `convert`: to move from one storage to another (from TarStorage to RedisStorage, for example)

Should these go into `sourmash-utils`?

**Comment by ctb** [comment]
maybe sourmash storage

?

On Wed, May 31, 2017 at 04:27:10PM -0700, Luiz Irber wrote:
> - `prefetch`: download all nodes beforehand (useful for IPFS)
> - `convert`: to move from one storage to another (from TarStorage to RedisStorage, for example)
> 
> Should these go into `sourmash-utils`?
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/dib-lab/sourmash/issues/261
-- 
C. Titus Brown, ctbrown@ucdavis.edu

**Comment by ctb** [comment]
with #1059 I'm wondering if it's actually worth implementing specific storage conversion commands, since we can "just" build a new database.

**Comment by luizirber** [comment]
> with #1059 I'm wondering if it's actually worth implementing specific storage conversion commands, since we can "just" build a new database using the old database as input.

That is true for `convert`, although `convert` could be smarter in some cases (avoid rebuilding internal nodes, just copy them, and so on).

I don't think #1059 deals with `prefetch`. The idea is mainly for IPFS support, but it would also fit with the #991 discussion around BDBag, and using the `fetch.txt` inside it to download the actual data. But, then again, there is `prepare`, which could also cover the `prefetch` case.

**Comment by ctb** [comment]
leaving this open but tagging with "plugin todo"
